src/python/pants/engine/target.py

```diff
@@ -1300,22 +1301,50 @@ def compute_value(cls, raw_value: Optional[bool], address: Address) -> Optional[
         return super().compute_value(raw_value, address)
 
 
+class ValidNumbers(Enum):
+    """What range of numbers are allowed for IntField and FloatField."""
+
+    positive_only = enum.auto()
+    positive_and_zero = enum.auto()
+    all = enum.auto()
+
+    def validate(self, num: float | int | None, alias: str, address: Address) -> None:
+        if num is None or self == self.all:  # type: ignore[comparison-overlap]
+            return
+        if self == self.positive_and_zero:  # type: ignore[comparison-overlap]
+            if num < 0:
+                raise InvalidFieldException(
+                    f"The {repr(alias)} field in target {address} must be >=0, but was set to `{num}`."
```

Comment:
Maybe replace `>=0` with narrative style "greater than or equal to zero"?

Reply:
Good idea!